### PR TITLE
Added pundit and configured user roles

### PR DESCRIPTION
**Pull request by the-b-boss**
In this feature branch:
- installed and configured Pundit gem (incl rescue from Pundit in application controller)
- defined methods for checking if a user is a premium or admin user in User.rb
- set default user role to "standard" in User.rb using after_initialize
- edited default userroles in application policy
- created a wiki_policy
- updated wiki-controller to authorize user permissions
- updated index and show views for wikis so that new/ edit/ delete buttons are only visible current user is authorized to perform action

### CODE REVIEW

**Comment by the-b-boss** [comment]
Happy with this branch - merged into master.

File: app/policies/application_policy.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+class ApplicationPolicy
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+  end
+
+  def index?
+    false
```
Comment (bjblock) [inline app/policies/application_policy.rb:10]: with this policy, are users ever able to get to the index?

File: app/policies/application_policy.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  def show?
+    scope.where(:id => record.id).exists?
```
Comment (bjblock) [inline app/policies/application_policy.rb:14]: what is this policy doing?  is it necessary?

File: app/policies/application_policy.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  def update?
+    user.present?
+  end
+
+  def edit?
+    update?
+  end
+
+  def destroy?
+    false
```
Comment (bjblock) [inline app/policies/application_policy.rb:34]: with this policy, are users ever able to destroy a wiki?

File: app/policies/wiki_policy.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class WikiPolicy < ApplicationPolicy
+
+  def index?
+    true
```
Comment (bjblock) [inline app/policies/wiki_policy.rb:4]: ah, you're overriding the ApplicationPolicy#index policy.  Ok, so users will be able to see the wiki index.  But what about destroy?

File: app/policies/application_policy.rb (hunk 1 above)
Comment (the-b-boss) [inline app/policies/application_policy.rb:10]: See separate wiki_policy that overrides this.

File: app/policies/application_policy.rb (hunk 2 above)
Comment (the-b-boss) [inline app/policies/application_policy.rb:14]: Not sure what this does - think it is just what gets generate automatically - was going to ask you what it is doing...

File: app/policies/application_policy.rb (hunk 3 above)
Comment (the-b-boss) [inline app/policies/application_policy.rb:34]: Was supposed to add this to the wiki_policy but see I didn't - will add in current branch and merge

File: app/policies/wiki_policy.rb (hunk 4 above)
Comment (the-b-boss) [inline app/policies/wiki_policy.rb:4]: will add in current branch and merge

**Pull request merged**